backend/fms_core/services/sample_next_step.py

```diff
@@ -256,7 +256,7 @@ def has_sample_completed_study(sample_obj: Sample, study_obj: Study) -> Tuple[Un
             errors.append(f"No step found for the given order.")
 
         # If the sample has completed the workflow, the StepHistory for the last step order in the study should
-        # have a workflow action of NEXT_STEP
+        # have a workflow action of NEXT_STEP or REPEAT_STEP
         if StepHistory.objects.filter(process_measurement__lineage__child=sample_obj, # for step with child
                                       study=study_obj, 
```

Comment:
Maybe we should reuse parts of the querysets? Like extract out querysets representing steps without children and with children, and then filter by workflow_action?

Reply:
not sure django optimizing is not really that obvious since most of the work building the query is done by the framework... We would have to benchmark the different options ... (i would personally instead possibly try to retrieve all step_history then parse them to make sure none if there is any is next_step or repeat_step ... but as far as i know most query will never actually execute the third and fourth option since they will be true after the first one ...